src/geocoding/tests/test_centroid.py

```diff
@@ -0,0 +1,156 @@
+import pytest
+from geocoding.centroid import centroid2d, maybe_centroid2d, geojson_centroid
+
+
+def test_centroid_of_none_should_fail():
+    with pytest.raises(TypeError):
+        centroid2d(None)
+
+
+def test_centroid_of_empty_list_should_fail():
+    with pytest.raises(ZeroDivisionError):
+        centroid2d([])
+
+
+def test_centroid_of_missing_coord_should_fail():
+    with pytest.raises(TypeError):
+        centroid2d([[None, 1], [1, 2]])
+
+
+def test_centroid_of_missing_coords_should_fail():
+    with pytest.raises(IndexError):
+        centroid2d([[], [1, 2]])
+
+
+def test_centroid_of_ill_typed_coord_should_fail():
+    with pytest.raises(TypeError):
+        centroid2d([[1, 2], [2, '']])
+
+
+@pytest.mark.parametrize('points', [
+    [[0, 0]], [[1, 0]], [[0, 1]], [[2, 2]]
+])
+def test_centroid_of_point_is_point_itself(points):
+    p = points[0]
+    assert p == centroid2d(points)
+
+
+@pytest.mark.parametrize('fixture', [
+    ([[0, 0], [2, 0]], [1, 0]),
+    ([[1, 1], [3, 3]], [2, 2]),
+    ([[1, 2], [2, 4], [3, 6]], [2, 4])
+])
+def test_centroid_of_list_of_points(fixture):
```

Comment:
I commented somewhere else, but note you can parametrize with more than one variable (instead of `fixture`, having `expected`, `obtained`. It'll improve legibility.

Reply:
thanks, done!